app.py

```diff
@@ -1,19 +1,26 @@
-from flask import Flask, request, jsonify, render_template, Response, send_from_directory
+from flask import Flask, request, jsonify, render_template, Response, send_from_directory, session
+from flask_session import Session
 from lib.db import dbconnect, db_create, School, User, Club, UserToClubMapping, Message, Position, UserClubPositionMapping
 
 import json
 import logging
-import hashlib
 import os
 
 app = Flask(__name__)
 
-db_options = {'db_file': 'my.db'}
 
+# config for Flask-Session
+SESSION_TYPE = 'filesystem'
+app.secret_key = '046b48a135cb6d14905c4bbd632bba23f4e56084877851e1bc2711d607a87d99'
+app.config.from_object(__name__)
+Session(app)
+
+db_options = {'db_file': 'my.db'}
 
-def get_or_create_school(session, school_name, address=None, email=None, phone=None):
+# must be "" or else IntegrityError: (sqlite3.IntegrityError) NOT NULL constraint failed
+def get_or_create_school(session, school_name, address="", email="", phone=""):
 
-    school = session.query(School).filter(School.name == school_name).one_or_none()
+    school = session.query(School).filter(School.address == address).one_or_none()
```

Comment:
i think this should filter on school name... if it actually should be address, address should not be an optional field